Support pam_umask by ensuring dokku user can access available plugins

Short summary: `/var/lib/dokku/plugins/available/letsencrypt` ownership is given to the root user and not to dokku

```
$ dokku version
dokku version 0.20.4
$ dokku plugin:install https://github.com/dokku/dokku-letsencrypt.git
This command must be run as root
$ sudo dokku plugin:install https://github.com/dokku/dokku-letsencrypt.git
-----> Cloning plugin repo https://github.com/dokku/dokku-letsencrypt.git to /var/lib/dokku/plugins/available/letsencrypt
Cloning into 'letsencrypt'...
remote: Enumerating objects: 74, done.
remote: Counting objects: 100% (74/74), done.
remote: Compressing objects: 100% (59/59), done.
remote: Total 535 (delta 38), reused 37 (delta 15), pack-reused 461
Receiving objects: 100% (535/535), 118.86 KiB | 746.00 KiB/s, done.
Resolving deltas: 100% (318/318), done.
-----> Plugin letsencrypt enabled
Adding user dokku to group adm
Starting nginx (via systemctl): nginx.service.
-----> Priming bash-completion cache
```

So far so good, then:
```
$ sudo dokku letsencrypt:help
 !     `letsencrypt:help` is not a dokku command.
 !     See `dokku help` for a list of available commands.
$ dokku plugin:list
plugn: 0.5.0
  00_dokku-standard    0.20.4 enabled    dokku core standard plugin
  20_events            0.20.4 enabled    dokku core events logging plugin
  app-json             0.20.4 enabled    dokku core app-json plugin
  apps                 0.20.4 enabled    dokku core apps plugin
  builder-dockerfile   0.20.4 enabled    dokku core builder-dockerfile plugin
  builder-herokuish    0.20.4 enabled    dokku core builder-herokuish plugin
  buildpacks           0.20.4 enabled    dokku core buildpacks plugin
  certs                0.20.4 enabled    dokku core certificate management plugin
  checks               0.20.4 enabled    dokku core checks plugin
  common               0.20.4 enabled    dokku core common plugin
  config               0.20.4 enabled    dokku core config plugin
  docker-options       0.20.4 enabled    dokku core docker-options plugin
  domains              0.20.4 enabled    dokku core domains plugin
  enter                0.20.4 enabled    dokku core enter plugin
  git                  0.20.4 enabled    dokku core git plugin
cat: /var/lib/dokku/plugins/available/letsencrypt/plugin.toml: Permission denied
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
main.TomlGet(0xc000010080, 0x2, 0x2)
	github.com/dokku/plugn/plugn.go:30 +0x2cd
github.com/progrium/go-basher.(*Context).HandleFuncs(0xc000080280, 0xc000010050, 0x5, 0x5, 0x0)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:195 +0x26d
github.com/progrium/go-basher.ApplicationWithPath(0xc000047e48, 0xc000047e78, 0x4, 0x4, 0x6300d8, 0x1, 0xc0000144c0, 0x19)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:77 +0x19c
github.com/progrium/go-basher.Application(0xc000047e48, 0xc000047e78, 0x4, 0x4, 0x6300d8, 0x664801)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:58 +0x1ce
main.main()
	github.com/dokku/plugn/plugn.go:102 +0x396
cat: /var/lib/dokku/plugins/available/letsencrypt/plugin.toml: Permission denied
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
main.TomlGet(0xc000010080, 0x2, 0x2)
	github.com/dokku/plugn/plugn.go:30 +0x2cd
github.com/progrium/go-basher.(*Context).HandleFuncs(0xc000080280, 0xc000010050, 0x5, 0x5, 0x0)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:195 +0x26d
github.com/progrium/go-basher.ApplicationWithPath(0xc000047e48, 0xc000047e78, 0x4, 0x4, 0x6300d8, 0x1, 0xc0000144c0, 0x19)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:77 +0x19c
github.com/progrium/go-basher.Application(0xc000047e48, 0xc000047e78, 0x4, 0x4, 0x6300d8, 0x664801)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:58 +0x1ce
main.main()
	github.com/dokku/plugn/plugn.go:102 +0x396
  letsencrypt                enabled    
  logs                 0.20.4 enabled    dokku core logs plugin
  network              0.20.4 enabled    dokku core network plugin
  nginx-vhosts         0.20.4 enabled    dokku core nginx-vhosts plugin
  plugin               0.20.4 enabled    dokku core plugin plugin
  proxy                0.20.4 enabled    dokku core proxy plugin
  ps                   0.20.4 enabled    dokku core ps plugin
  repo                 0.20.4 enabled    dokku core repo plugin
  resource             0.20.4 enabled    dokku core resource plugin
  scheduler-docker-local 0.20.4 enabled    dokku core scheduler-docker-local plugin
  shell                0.20.4 enabled    dokku core shell plugin
  ssh-keys             0.20.4 enabled    dokku core ssh-keys plugin
  storage              0.20.4 enabled    dokku core storage plugin
  tags                 0.20.4 enabled    dokku core tags plugin
  tar                  0.20.4 enabled    dokku core tar plugin
  trace                0.20.4 enabled    dokku core trace plugin
```
Of course:
```
$ ll /var/lib/dokku/plugins/available/
lrwxrwxrwx 1 dokku dokku   41 Jun 19 00:31 git -> /var/lib/dokku/core-plugins/available/git/
drwxr-x--- 5 root  root  4096 Jun 19 01:50 letsencrypt/
```
Running `sudo chown -R dokku:dokku letsencrypt/` fixes the issue (except for the bash completion which isn't updated)

Not sure if my server is in some way misconfigured or if it's an actual issue where cloning the plugin git repo runs as root instead of dokku...

After disabling/uninstalling/reinstalling the issue is still present:
```
$ sudo ls -lah /var/lib/dokku/plugins/available/letsencrypt
total 72K
drwxr-x--- 5 root  root  4.0K Jun 19 04:37 .
drwxr-xr-x 3 dokku dokku 4.0K Jun 19 04:37 ..
-rw-r----- 1 root  root   126 Jun 19 04:37 .editorconfig
drwxr-x--- 8 root  root  4.0K Jun 19 04:37 .git
-rw-r----- 1 root  root   129 Jun 19 04:37 .gitignore
-rw-r----- 1 root  root  1.1K Jun 19 04:37 LICENSE
-rw-r----- 1 root  root  9.7K Jun 19 04:37 README.md
-rwxr-x--- 1 root  root  1.5K Jun 19 04:37 commands
-rwxr-x--- 1 root  root   421 Jun 19 04:37 config
-rwxr-x--- 1 root  root   111 Jun 19 04:37 cron-job
-rwxr-x--- 1 root  root  5.6K Jun 19 04:37 functions
-rwxr-x--- 1 root  root   626 Jun 19 04:37 install
-rw-r----- 1 root  root   116 Jun 19 04:37 plugin.toml
drwxr-x--- 2 root  root  4.0K Jun 19 04:37 subcommands
drwxr-x--- 2 root  root  4.0K Jun 19 04:37 templates
ubuntu@vps-fda16d01 ~ 
$ sudo dokku letsencrypt:help
 !     `letsencrypt:help` is not a dokku command.
 !     See `dokku help` for a list of available commands.
$ dokku plugin:list
plugn: 0.5.0
  00_dokku-standard    0.20.4 enabled    dokku core standard plugin
  20_events            0.20.4 enabled    dokku core events logging plugin
  app-json             0.20.4 enabled    dokku core app-json plugin
  apps                 0.20.4 enabled    dokku core apps plugin
  builder-dockerfile   0.20.4 enabled    dokku core builder-dockerfile plugin
  builder-herokuish    0.20.4 enabled    dokku core builder-herokuish plugin
  buildpacks           0.20.4 enabled    dokku core buildpacks plugin
  certs                0.20.4 enabled    dokku core certificate management plugin
  checks               0.20.4 enabled    dokku core checks plugin
  common               0.20.4 enabled    dokku core common plugin
  config               0.20.4 enabled    dokku core config plugin
  docker-options       0.20.4 enabled    dokku core docker-options plugin
  domains              0.20.4 enabled    dokku core domains plugin
  enter                0.20.4 enabled    dokku core enter plugin
  git                  0.20.4 enabled    dokku core git plugin
cat: /var/lib/dokku/plugins/available/letsencrypt/plugin.toml: Permission denied
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
main.TomlGet(0xc000010080, 0x2, 0x2)
	github.com/dokku/plugn/plugn.go:30 +0x2cd
github.com/progrium/go-basher.(*Context).HandleFuncs(0xc000080280, 0xc000010050, 0x5, 0x5, 0x0)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:195 +0x26d
github.com/progrium/go-basher.ApplicationWithPath(0xc000047e48, 0xc000047e78, 0x4, 0x4, 0x6300d8, 0x1, 0xc0000144c0, 0x19)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:77 +0x19c
github.com/progrium/go-basher.Application(0xc000047e48, 0xc000047e78, 0x4, 0x4, 0x6300d8, 0x664801)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:58 +0x1ce
main.main()
	github.com/dokku/plugn/plugn.go:102 +0x396
cat: /var/lib/dokku/plugins/available/letsencrypt/plugin.toml: Permission denied
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
main.TomlGet(0xc000010080, 0x2, 0x2)
	github.com/dokku/plugn/plugn.go:30 +0x2cd
github.com/progrium/go-basher.(*Context).HandleFuncs(0xc000080280, 0xc000010050, 0x5, 0x5, 0x0)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:195 +0x26d
github.com/progrium/go-basher.ApplicationWithPath(0xc000047e48, 0xc000047e78, 0x4, 0x4, 0x6300d8, 0x1, 0xc0000144c0, 0x19)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:77 +0x19c
github.com/progrium/go-basher.Application(0xc000047e48, 0xc000047e78, 0x4, 0x4, 0x6300d8, 0x664801)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20200507184539-7aca7038b313/basher.go:58 +0x1ce
main.main()
	github.com/dokku/plugn/plugn.go:102 +0x396
  letsencrypt                enabled    
  logs                 0.20.4 enabled    dokku core logs plugin
  network              0.20.4 enabled    dokku core network plugin
  nginx-vhosts         0.20.4 enabled    dokku core nginx-vhosts plugin
  plugin               0.20.4 enabled    dokku core plugin plugin
  proxy                0.20.4 enabled    dokku core proxy plugin
  ps                   0.20.4 enabled    dokku core ps plugin
  repo                 0.20.4 enabled    dokku core repo plugin
  resource             0.20.4 enabled    dokku core resource plugin
  scheduler-docker-local 0.20.4 enabled    dokku core scheduler-docker-local plugin
  shell                0.20.4 enabled    dokku core shell plugin
  ssh-keys             0.20.4 enabled    dokku core ssh-keys plugin
  storage              0.20.4 enabled    dokku core storage plugin
  tags                 0.20.4 enabled    dokku core tags plugin
  tar                  0.20.4 enabled    dokku core tar plugin
  trace                0.20.4 enabled    dokku core trace plugin
```

--- a/plugins/plugin/install
+++ b/plugins/plugin/install
@@ -1,0 +1,13 @@
+#!/usr/bin/env bash
+set -eo pipefail
+[[ $DOKKU_TRACE ]] && set -x
+source "$PLUGIN_AVAILABLE_PATH/plugin/internal-functions"
+
+trigger-plugin-install() {
+  declare desc="plugin install plugin trigger"
+  declare trigger="install"
+
+  fn-plugin-set-permissions
+}
+
+trigger-plugin-install "$@"

--- a/plugins/plugin/internal-functions
+++ b/plugins/plugin/internal-functions
@@ -21,6 +21,12 @@ cmd-plugin-installed() {
   plugn list | awk '{print $1}' | grep -q "^${PLUGIN_NAME}$"
 }
 
+fn-plugin-set-permissions() {
+  declare desc="ensures correct permissions for all available plugins"
+  chown -R "${DOKKU_SYSTEM_USER}:${DOKKU_SYSTEM_GROUP}" "$PLUGIN_CORE_AVAILABLE_PATH" "$PLUGIN_AVAILABLE_PATH"
+  chown -R "${DOKKU_SYSTEM_USER}:${DOKKU_SYSTEM_GROUP}" "$PLUGIN_CORE_ENABLED_PATH" "$PLUGIN_ENABLED_PATH"
+}
+
 plugin_prime_bash_completion() {
   declare desc="primes the bash-completion cache"
 

--- a/plugins/plugin/update
+++ b/plugins/plugin/update
@@ -1,0 +1,13 @@
+#!/usr/bin/env bash
+set -eo pipefail
+[[ $DOKKU_TRACE ]] && set -x
+source "$PLUGIN_AVAILABLE_PATH/plugin/internal-functions"
+
+trigger-plugin-update() {
+  declare desc="plugin update plugin trigger"
+  declare trigger="update"
+
+  fn-plugin-set-permissions
+}
+
+trigger-plugin-update "$@"

--- a/tests/unit/plugin.bats
+++ b/tests/unit/plugin.bats
@@ -180,3 +180,69 @@ teardown() {
   echo "status: $status"
   assert_failure
 }
+
+@test "(plugin) plugin:install permissions set properly" {
+  run /bin/bash -c "ls -lah /var/lib/dokku/core-plugins/available/git/commands"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+  assert_output_contains "root" "0"
+  assert_output_contains "dokku" "3"
+
+  run /bin/bash -c "chown -R root:root /var/lib/dokku/core-plugins/available/git"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+
+  run /bin/bash -c "ls -lah /var/lib/dokku/core-plugins/available/git/commands"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+  assert_output_contains "root" "2"
+  assert_output_contains "dokku" "1"
+
+  run /bin/bash -c "dokku plugin:install"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+
+  run /bin/bash -c "ls -lah /var/lib/dokku/core-plugins/available/git/commands"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+  assert_output_contains "root" "0"
+  assert_output_contains "dokku" "3"
+}
+
+@test "(plugin) plugin:update permissions set properly" {
+  run /bin/bash -c "ls -lah /var/lib/dokku/core-plugins/available/git/commands"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+  assert_output_contains "root" "0"
+  assert_output_contains "dokku" "3"
+
+  run /bin/bash -c "chown -R root:root /var/lib/dokku/core-plugins/available/git"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+
+  run /bin/bash -c "ls -lah /var/lib/dokku/core-plugins/available/git/commands"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+  assert_output_contains "root" "2"
+  assert_output_contains "dokku" "1"
+
+  run /bin/bash -c "dokku plugin:update"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+
+  run /bin/bash -c "ls -lah /var/lib/dokku/core-plugins/available/git/commands"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+  assert_output_contains "root" "0"
+  assert_output_contains "dokku" "3"
+}

